src/streaming/text/TextSourceBuffer.js

```diff
@@ -326,7 +330,10 @@ function TextSourceBuffer() {
                             subOffset += sample.subSizes[j];
                         }
                         try {
-                            result = parser.parse(ccContent, sampleStart / timescale, (sampleStart + sample.duration) / timescale, images);
+                            // check if we must apply an offset to ttml time
+                            let manifest = manifestModel.getValue();
```

Comment:
Please add a comment here that MSS has an offset corresponding to the segment start, while DASH has none.